pkg/controller/usersshkeysagent/usersshkeys_controller.go

```diff
@@ -215,32 +220,41 @@ func createBuffer(data map[string][]byte) (*bytes.Buffer, error) {
 	return buffer, nil
 }
 
-func updateOwnAndPermissions(path string) error {
-	sshPath := strings.TrimSuffix(path, "/authorized_keys")
-	if err := os.Chmod(sshPath, os.FileMode(0700)); err != nil {
-		return fmt.Errorf("failed to change permission on file: %v", err)
-	}
+func ChownR(path string, uid, gid int) error {
+	return filepath.Walk(path, func(name string, info os.FileInfo, err error) error {
+		if err == nil {
+			err = os.Chown(name, uid, gid)
+		}
+		return err
+	})
+}
 
-	if err := os.Chmod(path, os.FileMode(0600)); err != nil {
-		return fmt.Errorf("failed to change permission on file: %v", err)
+func createFileIfNotExists(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		_, err := os.Create(path)
+		if err != nil {
+			return err
+		}
 	}
 
-	userHome := strings.TrimSuffix(sshPath, "/.ssh")
-	fileInfo, err := os.Stat(userHome)
+	return nil
+}
+
+func updateOwnAndPermissions(path string) error {
+	paths := strings.Split(path, "/")
+
+	fileDir := strings.Join(paths[:len(paths)-1], "/")
+	fileInfo, err := os.Stat(fileDir)
 	if err != nil {
-		return fmt.Errorf("failed describing the authorized_keys file in path %s: %v", userHome, err)
+		return fmt.Errorf("failed describing the directory in path %s: %v", fileDir, err)
 	}
 
 	uid := fileInfo.Sys().(*syscall.Stat_t).Uid
 	gid := fileInfo.Sys().(*syscall.Stat_t).Gid
 
-	if err := os.Chown(path, int(uid), int(gid)); err != nil {
+	if err := ChownR(path, int(uid), int(gid)); err != nil {
```

Comment:
Is it necessary to update the permissions for the directory @irozzo-1A ?

Kindly give it a look if it's fine